### PR TITLE
Lesson03

### PR DESCRIPTION
**Pull request by uidev1219**
# [Lesson No.3](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#3)
## Concerns and areas to focus on
- I'm worried about naming variables and functions.
- For the sake of clarity, I've separated the creation of additional elements from the addition of elements to the ul, but what do you think?

### CODE REVIEW

File: lesson03/src/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  img.src = `/img/${data.src}`;
+  const text = document.createTextNode(data.text)
+  li.appendChild(a).appendChild(img).appendChild(text);
+  return li;
+}
+
+const elems = dataList.map(createImageList);
+
+const ul = document.getElementById('js-list');
+elems.map((elem) => ul.appendChild(elem));
```
Comment (443n0511) [inline lesson03/src/index.js:28]: IMO
For this process, I learned [Document.createDocumentFragment()](https://developer.mozilla.org/ja/docs/Web/API/Document/createDocumentFragment) from Mr.Morita. I thought this was better in terms of performance than creating a function and looping through the map.

File: lesson03/src/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const a = document.createElement('a')
+  a.href = data.href;
+  const img = document.createElement('img')
+  img.src = `/img/${data.src}`;
+  const text = document.createTextNode(data.text)
+  li.appendChild(a).appendChild(img).appendChild(text);
+  return li;
+}
+
+const elems = dataList.map(createImageList);
```
Comment (shouyamamoto) [inline lesson03/src/index.js:25]: I thought it was great that you carved out a function to run in this map! ✊

File: lesson03/src/index.js (hunk 1 above)
Comment (shouyamamoto) [inline lesson03/src/index.js:28]: IMO
For the `(elem) => ul.appendChild(elem)` part, you might want to add a fragment as Kimura says, and also cut out the process as in Line 25.

**Comment by 443n0511** [comment]
スクリーンショット 2021-08-05 12 48 54
It seems that the output is not correct.

**Comment by uidev1219** [comment]
Oh! I missed!
Thanks for  pointing it out

I fixed!! https://github.com/uidev1219/frontendHandson/pull/3/commits/aa686e29663fb852e5e3d286d17cd99eb078027f

**Review by 443n0511 — APPROVED** [review]

**Review by shouyamamoto — APPROVED** [review]

**Pull request merged**